build.rs

```diff
@@ -1,288 +1,237 @@
+        .call()
+        .expect("Failed to download Speech SDK")
+        .into_reader();
+    while let Some(mut zip_file) = zip::read::read_zipfile_from_stream(&mut sdk).unwrap() {
+        if zip_file.is_dir() {
+            fs::create_dir_all(path.join(zip_file.enclosed_name().unwrap())).unwrap();
+        } else {
+            let file_path = path.join(zip_file.enclosed_name().unwrap());
+            fs::create_dir_all(file_path.parent().unwrap()).unwrap();
+            let mut file = fs::File::create(file_path).unwrap();
+            std::io::copy(&mut zip_file, &mut file).unwrap();
+        }
     }
+}
 
-    if renew {
-        let reader = File::open(sdk_zip_file).unwrap();
-        let mut archive = ZipArchive::new(BufReader::new(reader)).unwrap();
-
-        for i in 0..archive.len() {
-            let mut file = archive.by_index(i).unwrap();
-            let outpath = sdk_output_dir.join(file.mangled_name());
+struct LinkParams {
+    lib_arg: String,
+    search_arg: String,
+    include_dir: Option<PathBuf>,
+}
 
-            if file.name().ends_with('/') {
-                std::fs::create_dir_all(&outpath).unwrap();
+/// Get the link parameters for the Speech SDK based on the target OS and architecture.
+fn link_params(sdk_dir: &Path) -> LinkParams {
+    match TargetOs::get() {
+        TargetOs::Linux => {
+            let mut lib_dir = sdk_dir.join("lib");
+            if cfg!(target_arch = "x86") {
+                lib_dir.push("x86")
+            } else if cfg!(target_arch = "x86_64") {
+                lib_dir.push("x64")
+            } else if cfg!(target_arch = "arm") {
+                lib_dir.push("arm32")
+            } else if cfg!(target_arch = "aarch64") {
+                lib_dir.push("arm64")
             } else {
-                if let Some(p) = outpath.parent() {
-                    if !p.exists() {
-                        std::fs::create_dir_all(p).unwrap();
-                    }
-                }
-                let mut outfile = std::fs::File::create(&outpath).unwrap();
-                std::io::copy(&mut file, &mut outfile).unwrap();
+                panic!("Unsupported target architecture; only x86, x86_64, arm, and aarch64 are supported on Linux.")
+            }
+            LinkParams {
+                lib_arg: "dylib=Microsoft.CognitiveServices.Speech.core".to_owned(),
+                search_arg: format!("native={}", lib_dir.to_str().unwrap()),
+                include_dir: Some(sdk_dir.join("include/c_api")),
+            }
+        }
+        TargetOs::MacOS => LinkParams {
+            lib_arg: "framework=MicrosoftCognitiveServicesSpeech".to_owned(),
+            search_arg: format!(
+                "framework={}",
+                sdk_dir.join("macos-arm64_x86_64").to_str().unwrap()
+            ),
+            include_dir: Some(
+                sdk_dir
+                    .join("macos-arm64_x86_64/MicrosoftCognitiveServicesSpeech.framework/Headers"),
```

Comment:
I'm getting this bindgen error when testing on macOS:
```bash
--- stderr
  c_api/wrapper.h:7:10: fatal error: 'speechapi_c.h' file not found
  thread 'main' panicked at build.rs:101:10:
  Unable to generate bindings: ClangDiagnostic("c_api/wrapper.h:7:10: fatal error: 'speechapi_c.h' file not found\n")
```
The `sdk_output` has the following dir structure:

```bash
$ cd ./target/debug/build/cognitive-services-speech-sdk-rs-f69e2a90dd5248fa/out/sdk_output
$ tree                             
.
├── LICENSE.md
├── MicrosoftCognitiveServicesSpeech.xcframework
│   ├── Info.plist
│   └── macos-arm64_x86_64
│       └── MicrosoftCognitiveServicesSpeech.framework
│           ├── Headers
│           ├── MicrosoftCognitiveServicesSpeech
│           ├── Modules
│           ├── Resources
│           └── Versions
│               ├── A
│               │   ├── Headers
│               │   │   ├── SPXAudioConfiguration.h
│               │   │   ├── SPXAudioDataStream.h
│               │   │   ├── SPXAudioProcessingOptions.h
│               │   │   ├── SPXAudioStream.h
│               │   │   ├── SPXAudioStreamFormat.h
```
The `macos-arm64_x86_64/MicrosoftCognitiveServicesSpeech.framework/Headers` should be a symlink(or `alias` in macOS terms), but instead, it is unzipped into a text file with the following content:
```
Versions/Current/Headers
```
this is why `bindgen` complains about `'speechapi_c.h' file not found`. 

Maybe on macOS we should change the `include_dir` path to 
```rust
Some(sdk_dir
    .join("macos-arm64_x86_64/MicrosoftCognitiveServicesSpeech.framework/Versions/A/Headers"),
),
```
or use the `unzip` command line utility to extract the SDK zip file instead?

Reply:
just tested on my Mac and facing same issue

Reply:
I *think* the issue is that I wasn't setting the file mode, I've changed the code to do that now, can you try again and see if the symlinks are now actually symlinks?